src/main/kotlin/com/bugsnag/android/gradle/BugsnagPlugin.kt

```diff
@@ -186,11 +187,14 @@ class BugsnagPlugin : Plugin<Project> {
             val manifestInfoFileProvider = registerManifestUuidTask(project, variant)
             val mappingFilesProvider = createMappingFileProvider(project, variant, output)
 
+            val versionCodeProvider = DefaultProvider { output.versionCode }
```

Comment:
You should use providerfactory instead of internal APIs

```suggestion
            val versionCodeProvider = project.providers.provider { output.versionCode }
```